app/bundles/CoreBundle/Helper/ListParser/ArrayListParser.php

```diff
@@ -12,11 +12,7 @@ public function parse($list): array
             throw new FormatNotSupportedException();
         }
 
-        if (
-            array_key_exists(0, $list)
-            && !is_array($list[0])
-            && array_keys($list) === range(0, count($list) - 1)
-        ) {
+        if (!array_filter(array_keys($list), 'is_string') && !array_filter($list, 'is_array')) {
```

Comment:
If you hit add several times you have inputs for `[0]`, `[1]`, and `[2]`. Hitting remove on the first two leaves `[2]`. Not sure if the intention is Javascript should handle it but I figured this code in here for `ArrayListParser` should safely be able to handle all-numerical keys with no arrays as values as an `ArrayList` - instead of relying on index 0 being present and the keys all being incremental. Can't imagine you'd ever get an object like that with numeric keys but non-consecutive that would need supporting... And this means the JS can be simpler like it is which is just to drop out and not worry about renumbering the inputs